client/my-sites/stats/hooks/use-stats-purchases.tsx

```diff
@@ -121,24 +123,23 @@ export default function useStatsPurchases( siteId: number | null ) {
 	const hasLoadedSitePurchases = useSelector( hasLoadedSitePurchasesFromServer );
 
 	// Determine whether a product is owned.
-	// TODO we need to do plan check as well, because Stats products would be built into other plans.
-	const isFreeOwned = useMemo( () => {
-		return isProductOwned( sitePurchases, PRODUCT_JETPACK_STATS_FREE );
-	}, [ sitePurchases ] );
+	const isFreeOwned = useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, FEATURE_STATS_FREE );
+	} );
 
 	const isCommercialOwned = useMemo(
 		() => isCommercialPurchaseOwned( sitePurchases ),
 		[ sitePurchases ]
 	);
 
+	// Pay what you want
 	const isPWYWOwned = useMemo( () => {
 		return isProductOwned( sitePurchases, PRODUCT_JETPACK_STATS_PWYW_YEARLY );
 	}, [ sitePurchases ] );
 
-	const supportCommercialUse = useMemo(
-		() => supportCommercialPurchaseUse( sitePurchases ),
-		[ sitePurchases ]
-	);
+	const supportCommercialUse = useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, FEATURE_STATS_COMMERCIAL );
+	} );
 
 	const isLegacyCommercialLicense = useMemo( () => {
 		const purchases = filterPurchasesByProducts( sitePurchases, [
```

Comment:
Which feature should be used here? Or is the reliance on products purchased fine here?